### ISSUE TITLE
[BUG]  "Counts conversion factor" and "Flux scaling" are 0 by default instead of 1

### ISSUE THREAD

**Issue by eteq** [spacetelescope/jdaviz#2550]
### Jdaviz component

Imviz

### Description

"Counts conversion factor" and "Flux scaling" options of the aperture photometry plugin are 0 when you open up apphot.  Both, though, seem like they should be `1` not `0` to be the case of "ignore those correction factors".

### How to Reproduce

1. Install jdaviz 
2. Open up aperture photometry plugin

### Expected behavior

I would have expected these two scale factors to be 1 as the no-op case, since 0 should lead to divide-by-zero errrors.  It does not, so I guess `0` must be a placeholder for "ignore", but that's confusing.

### Jupyter

IPython          : 8.16.0
ipykernel        : 6.25.2
ipywidgets       : 8.1.1
jupyter_client   : 7.0.6
jupyter_core     : 5.3.2
jupyter_server   : 1.24.0
jupyterlab       : 4.0.6
nbclient         : 0.8.0
nbconvert        : 7.8.0
nbformat         : 5.9.2
notebook         : 7.0.4
qtconsole        : 5.4.3
traitlets        : 5.10.1

### Software versions

Python 3.11.5 (main, Sep  2 2023, 14:16:33) [GCC 13.2.1 20230801]
>>> import numpy; print("Numpy", numpy.__version__)
Numpy 1.26.0
>>> import astropy; print("astropy", astropy.__version__)
astropy 5.3.1
>>> import matplotlib; print("matplotlib", matplotlib.__version__)
matplotlib 3.8.0
>>> import scipy; print("scipy", scipy.__version__)
scipy 1.11.3
>>> import skimage; print("scikit-image", skimage.__version__)
scikit-image 0.21.0
>>> import asdf; print("asdf", asdf.__version__)
asdf 2.15.0
>>> import stdatamodels; print("stdatamodels", stdatamodels.__version__)
stdatamodels 1.6.0
>>> import gwcs; print("gwcs", gwcs.__version__)
gwcs 0.18.3
>>> import regions; print("regions", regions.__version__)
regions 0.7
>>> import specutils; print("specutils", specutils.__version__)
specutils 1.11.0
>>> import specreduce; print("specreduce", specreduce.__version__)
specreduce 1.3.0
>>> import photutils; print("photutils", photutils.__version__)
photutils 1.8.0
>>> import astroquery; print("astroquery", astroquery.__version__)
astroquery 0.4.6
>>> import yaml; print("pyyaml", yaml.__version__)
pyyaml 6.0.1
>>> import asteval; print("asteval", asteval.__version__)
asteval 0.9.30
>>> import idna; print("idna", idna.__version__)
idna 3.4
>>> import traitlets; print("traitlets", traitlets.__version__)
traitlets 5.10.1
>>> import bqplot; print("bqplot", bqplot.__version__)
bqplot 0.12.39
>>> import bqplot_image_gl; print("bqplot-image-gl", bqplot_image_gl.__version__)
bqplot-image-gl 1.4.11
>>> import glue; print("glue-core", glue.__version__)
glue-core 1.12.0
>>> import glue_jupyter; print("glue-jupyter", glue_jupyter.__version__)
glue-jupyter 0.17.0
>>> import glue_astronomy; print("glue-astronomy", glue_astronomy.__version__)
glue-astronomy 0.10.0
>>> import echo; print("echo", echo.__version__)
echo 0.8.0
>>> import ipyvue; print("ipyvue", ipyvue.__version__)
ipyvue 1.9.2
>>> import ipyvuetify; print("ipyvuetify", ipyvuetify.__version__)
ipyvuetify 1.8.10
>>> import ipysplitpanes; print("ipysplitpanes", ipysplitpanes.__version__)
ipysplitpanes 0.2.0
>>> import ipygoldenlayout; print("ipygoldenlayout", ipygoldenlayout.__version__)
ipygoldenlayout 0.4.0
>>> import ipypopout; print("ipypopout", ipypopout.__version__)
ipypopout 1.0.0
>>> import jinja2; print("Jinja2", jinja2.__version__)
Jinja2 3.1.2
>>> import voila; print("voila", voila.__version__)
voila 0.4.1
>>> import vispy; print("vispy", vispy.__version__)
vispy 0.13.0
>>> import sidecar; print("sidecar", sidecar.__version__)
sidecar 0.5.2
>>> import jdaviz; print("Jdaviz", jdaviz.__version__)
Jdaviz 3.5.0
>>>

**Comment by pllim** [comment]
There is actually a subtle difference between 1 and 0, and 0 is the default in the code on purpose. This is not a bug. If you look at Point 8 in https://jdaviz.readthedocs.io/en/latest/imviz/plugins.html#aperture-photometry , it says, "Setting this to 1 is equivalent to not applying any scaling. This field is only used if data has a valid unit. If this field is not applicable for you, leave it at 0."

tl;dr 

* 0 (default) -- Do not even convert to magnitude.
* 1 -- Convert to magnitude but without any zeropoint scaling (usually this is raw instrumental magnitude).

This is because I want people to stop and think about what magnitude system they really want. I do not want people to blindly take an instrumental mag and think it is STMAG and so on.

**Comment by pllim** [comment]
Same deal with counts conversion factor. Counts can be ambiguous, so 0 means I do not do it at all. And if you request for it to be done, you better know what you are doing.

**Comment by kecnry** [comment]
Would it be more clear if we had a switch to enable/disable and if enabled, _then_ you can access changing the value?

**Comment by camipacifici** [comment]
I understand Erik's concern and I kind of agree (although it did not hit me before).
How about N/A (or some equivalent) instead of 0 when it is not in use?

**Comment by pllim** [comment]
Then we have to define it as "Any" instead of "Float", which is more processing for that one thing.

**Comment by kecnry** [comment]
Let's reopen this then, I think there is definitely some improvements we could make, right?

**Comment by pllim** [comment]
Once you decide how you want this done, please update the title and open a JIRA ticket. Thanks!
